lib/src/exceptions/posthog_error_tracking_autocapture_integration.dart

```diff
@@ -0,0 +1,175 @@
+  void stop() {
+    if (!_isEnabled) return;
+
+    _isEnabled = false;
+
+    // Restore original handlers
+    if (_originalFlutterErrorHandler != null) {
+      FlutterError.onError = _originalFlutterErrorHandler;
+      _originalFlutterErrorHandler = null;
+    }
+
+    if (_originalPlatformErrorHandler != null) {
+      PlatformDispatcher.instance.onError = _originalPlatformErrorHandler;
+      _originalPlatformErrorHandler = null;
+    }
+  }
+
+  /// Flutter framework error handler
+  void _setupFlutterErrorHandler() {
+    // prevent circular calls
+    if (FlutterError.onError == _posthogFlutterErrorHandler) {
+      return;
+    }
+
+    _originalFlutterErrorHandler = FlutterError.onError;
+
+    FlutterError.onError = _posthogFlutterErrorHandler;
+  }
+
+  void _posthogFlutterErrorHandler(FlutterErrorDetails details) {
+    // don't capture silent errors (could maybe be a config?)
+    if (!details.silent) {
+      _captureException(
+        error: details.exception,
+        stackTrace: details.stack,
+        handled: false,
+        context: details.context?.toString(),
+      );
+    }
+
+    // Call the original handler
+    if (_originalFlutterErrorHandler != null) {
+      try {
+        _originalFlutterErrorHandler!(details);
+      } catch (e) {
+        // Pretty sure we should be doing this to avoid infinite loops
+        debugPrint(
+            'PostHog: Error in original FlutterError.onError handler: $e');
+      }
+    } else {
+      // If no original handler, use the default behavior (default is to dump to console)
+      FlutterError.presentError(details);
+    }
+  }
+
+  /// Platform error handler for Dart runtime errors
+  void _setupPlatformErrorHandler() {
+    // prevent circular calls
+    if (PlatformDispatcher.instance.onError == _posthogPlatformErrorHandler) {
+      return;
+    }
+
+    _originalPlatformErrorHandler = PlatformDispatcher.instance.onError;
+    PlatformDispatcher.instance.onError = _posthogPlatformErrorHandler;
+  }
+
+  bool _posthogPlatformErrorHandler(Object error, StackTrace stackTrace) {
+    _captureException(
+        error: error,
+        stackTrace: stackTrace,
+        handled: false,
+        context: 'Platform error');
+
+    // Call the original handler
+    if (_originalPlatformErrorHandler != null) {
+      try {
+        return _originalPlatformErrorHandler!(error, stackTrace);
+      } catch (e) {
+        debugPrint(
+            'PostHog: Error in original PlatformDispatcher.onError handler: $e');
+        return true; // Consider the error handled
+      }
+    }
+
+    return false; // No original handler, don't modify behavior
+  }
+
+  Future<void> _captureException({
+    required dynamic error,
+    required StackTrace? stackTrace,
+    required bool handled,
+    String? context,
+  }) {
+    return _posthog.captureException(
+      error: error,
+      stackTrace: stackTrace ?? StackTrace.current,
```

Comment:
do not use `StackTrace.current` here, you need to figure this out inside of the captureException so you can set syntheetic or not